src/main/scala/io/shiftleft/js2cpg/parser/JsSource.scala

```diff
@@ -83,6 +83,8 @@ class JsSource(val srcDir: File, val projectDir: Path, val source: Source) {
   }
 
   private def constructSourceFilePath(sourceFileName: String): File = sourceFileName match {
+    case _ if sourceFileName.isEmpty =>
```

Comment:
just a nit-pick, but I'd either write this as 
```suggestion
    case "" =>
```

... or change the `match` to an `if-else` chain if we don't actually want to use it for matching any patterns